sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -578,6 +616,45 @@ def split_next(self):
 
         return left_child_node, right_child_node
 
+    def _compute_interactions(self, node):
+        r"""Compute features allowed by interactions to be inherited by child nodes.
+
+        Example: Assume constraints [{0, 1}, {1, 2}].
+           1      <- Both constraint groups could be applied from now on
+          / \
+         1   2    <- Left split still fulfills both constraint groups.
+        / \ / \      Right split at feature 2 has only group {1, 2} from now on.
+
```

Comment:
Since this is docstring is for developers:

```suggestion
        LightGBM uses the same logic for overlapping groups. See
        https://github.com/microsoft/LightGBM/issues/4481 for details.
```